lib/dynflow/executors/abstract/core.rb

```diff
@@ -35,7 +35,12 @@ def handle_event(event)
           handle_work(@director.handle_event(event))
         end
 
-        def work_finished(work)
+        def plan_events(delayed_events)
+          delayed_events.each { |event| @world.plan_event(event) }
```

Comment:
I put a pry into the loop

```ruby
> event
=> "Dynflow::Dispatcher::Event[execution_plan_id: 773011e9-a6a6-4929-b3c1-96992e14126f, step_id: 2, event: tick, time: ]"

> event.class
String
```

looks like the event gets `to_s` and deserialization is being skipped somewhere.

Reply:
Added `Array` serialization to the serializer.

Reply:
```
E, [2019-10-07T12:30:17.954927 #1] ERROR -- /parallel-executor-core: wrong number of arguments (given 1, expected 4..5) (ArgumentError)
/data/lib/dynflow/world.rb:230:in 'plan_event'
/data/lib/dynflow/executors/abstract/core.rb:39:in 'block in plan_events'
/data/lib/dynflow/executors/abstract/core.rb:39:in 'each'
/data/lib/dynflow/executors/abstract/core.rb:39:in 'plan_events'
/data/lib/dynflow/executors/abstract/core.rb:43:in 'work_finished'
```

Reply:
I wrote (and it still hurts) some tests for it, so we have it at least a bit covered.